GNUmakefile

```
@@ -152,7 +154,26 @@ dev-docker: linux
 	@docker pull consul:$(CONSUL_IMAGE_VERSION) >/dev/null
 	@echo "Building Consul Development container - $(CONSUL_DEV_IMAGE)"
 	#  'consul:local' tag is needed to run the integration tests
-	@DOCKER_DEFAULT_PLATFORM=linux/amd64 docker build $(NOCACHE) $(QUIET) -t '$(CONSUL_DEV_IMAGE)' -t 'consul:local' --build-arg CONSUL_IMAGE_VERSION=$(CONSUL_IMAGE_VERSION) $(CURDIR)/pkg/bin/linux_amd64 -f $(CURDIR)/build-support/docker/Consul-Dev.dockerfile
+	@docker buildx create --use && docker buildx build -t 'consul:local' \
```

This will actually cause the integration tests to not work because the image cache for buildx images is not the regular system cache and the integration tests don't use this buildx cache right now. You can test this out by running:

```
docker rmi consul-dev
docker rmi consul:local
```

And then running `make make test-envoy-integ`. It should fail with "image not found in registry" errors.

That's probably okay, but we'll need a new target for the integration tests to depend on (right now they depend on dev-docker) that builds using regular docker build.

---

@lkysow I'm doing some experiments on the side w/ skaffold and I believe the following setup might work better for dev environments. It would fix the scoping issue you mentioned and also increase performance with caching 
```suggestion
	@docker buildx use default && docker buildx build -t 'consul:local' \
```